railties/test/application/test_runner_test.rb

```diff
@@ -502,15 +502,15 @@ def test_output_inline_by_default
       create_test_file :models, "post", pass: false, print: false
 
       output = run_test_command("test/models/post_test.rb")
-      expect = %r{Running:\n\nF\n\nFailure:\nPostTest#test_truth \[[^\]]+test/models/post_test.rb:6\]:\nwups!\n\nrails test test/models/post_test.rb:4\n\n\n\n}
```

Comment:
Hey @dhh I was checking this out, and it seems related to minitest-reporters gem. Removing the gem and running the test make most tests pass.

What happened is that on this branch, minitest ended up loading minitest-reporters plugin first. Making the following code https://github.com/rails/rails/blob/main/railties/lib/minitest/rails_plugin.rb#L61-L69 not load  `Rails::TestUnitReporter`. 

There are a [few reports](https://github.com/minitest-reporters/minitest-reporters/issues/247) about similar situations. 

One way to fix is to remove that gem and rework the JUnit artifacts on CI without it. minitest-reporters was added on #35698 in order to generate those junit reports only. Would this be a good way forward?

The other option is making `railties/lib/minitest/rails_plugin.rb` work well with minitest-reporters.